lib/client.go

```diff
@@ -191,7 +194,7 @@ func (c *Client) GetCAInfo(req *api.GetCAInfoRequest) (*GetCAInfoResponse, error
 }
 
 // GenCSR generates a CSR (Certificate Signing Request)
-func (c *Client) GenCSR(req *api.CSRInfo, id string) ([]byte, bccsp.Key, error) {
+func (c *Client) GenCSR(req *api.CSRInfo, id string, k bccsp.Key) ([]byte, bccsp.Key, error) {
```

Comment:
Backwards compatibility isn't necessarily a strong requirement but this is the client API. Do we need (or really want) to make this change to add an argument and overload the semantics based on the presence of the key?

Reply:
Fixed